src/orchard/query_environment.clj

```diff
@@ -0,0 +1,67 @@
+(ns orchard.query-environment
```

Comment:
Perhaps `query-repl`, `repl-state` or `inspect-repl`? I'm afraid for many people "environment" might mean a shell for instance. 

The namespace should also have some docstring.

Reply:
I'm not keen on the name either, I'll look for something existing. REPL has too many connotations with the command line prompt, and not enough with clojure's state.